backend/drm/drm.c

```diff
@@ -566,6 +566,31 @@ static bool drm_connector_set_mode(struct wlr_output *output,
 	return false;
 }
 
+
+bool wlr_drm_connector_add_mode(struct wlr_output *output, drmModeModeInfo *modeinfo) {
+	struct wlr_drm_connector *conn = (struct wlr_drm_connector *)output;
+	struct wlr_drm_mode *mode = calloc(1, sizeof(*mode));
+
+	assert(modeinfo);
+	if (!mode || modeinfo->type != DRM_MODE_TYPE_USERDEF) {
```

Comment:
This leaks `mode` in case `modeinfo->type` is incorrect. We can probably move the `modeinfo->type` check before allocating.